### PR TITLE
Avoid head -c GNUism in reftests

### PR DESCRIPTION
**Pull request by madroach**
`head -c` is not posix-compliant, use `cut -b` instead.

### CODE REVIEW

File: master_changes.md [hunk 1]
```diff
@@ -144,6 +144,7 @@ users)
   * env tests: add regression test for append/prepend operators to empty environment variables [#5925, #5935 @dra27]
   * env.win32: add regression test for handling the empty entry in PATH-like variables [#5926, #5935 @dra27]
   * lint: add W41 examples [#5927 @dra27]
+  * [BUG]: head -c is not posix compliant. Use cut -b instead.
```
Comment (kit-ty-kate) [inline master_changes.md:147]:
```suggestion
  * [BUG]: head -c is not posix compliant. Use cut -b instead. [#5989 @madroach]
```

**Review by kit-ty-kate — APPROVED** [review]

**Review by rjbou — APPROVED** [review]
thanks!

**Review by rjbou — CHANGES_REQUESTED** [review]
There is a missing one in list.unix.test

**Comment by madroach** [comment]
> There is a missing one in list.unix.test

This one uses `head -n 1`, which is POSIX conforming. It's the `-c 1` parameter, which is not POSIX conforming.

**Review by rjbou — APPROVED** [review]

**Comment by kit-ty-kate** [comment]
Thanks a lot!
